File: internal/indexer/testdata/python/defs_b.py
from typing import List


# AppProcessor handles data processing.
class AppProcessor:
    def __init__(self, name: str, workers: int = 1):
        self.name = name
        self.workers = workers

    def run(self):
        for i in range(self.workers):
            print(f"Worker {i} started for {self.name}")

    def stop(self):
        print(f"Stopping {self.name}")


def format_data(items: List[str]) -> str:
    return ", ".join(items)

File: internal/indexer/testdata/js/defs_b.js
/**
 * AppProcessorJs handles data processing (JS).
 */
class AppProcessorJs {
  constructor(name, workers = 1) {
    this.name = name;
    this.workers = workers;
  }

  run() {
    for (let i = 0; i < this.workers; i++) {
      console.log(`Worker ${i} started for ${this.name}`);
    }
  }

  stop() {
    console.log(`Stopping ${this.name}`);
  }
}

module.exports = { AppProcessorJs };
